app/grandchallenge/reader_studies/views.py

```diff
@@ -599,10 +599,14 @@ def form_valid(self, form):  # noqa: C901
             },
         )
         rs.add_editor(self.request.user)
-        if form.cleaned_data["copy_images"]:
-            rs.images.set(reader_study.images.all())
-        if form.cleaned_data["copy_hanging_list"]:
-            rs.hanging_list = reader_study.hanging_list
+        if form.cleaned_data["copy_display_sets"]:
+            for ds in reader_study.display_sets.all():
+                new_ds = DisplaySet.objects.create(reader_study=rs)
+                new_ds.values.set(ds.values.all())
```

Comment:
Could this be moved to an asynchronous task? This operation will take a while (see, migrating studies to display sets) and we will have problems with users getting timeout errors and then trying again, copying the study many times. Add a success message to the response to let the user know that the task will be done asynchronously. Should be on the long tasks queue, using an atomic block.